files/collect_api_events.rb

```diff
@@ -16,7 +16,13 @@
 lockdir   = ARGV[2] || '/opt/puppetlabs/common_events/cache/state'
 
 def main(confdir, logpath, lockdir)
-  log = CommonEvents::Logger.new(logpath)
+  settings = YAML.safe_load(File.read("#{confdir}/events_collection.yaml"))
+
+  if settings['log_rotation'] == '0'
+    log = CommonEvents::Logger.new(logpath, settings['log_rotation'].to_i)
+  elsif (log = CommonEvents::Logger.new(logpath, settings['log_rotation'].downcase))
```

Comment:
In general you don't want the conditional clause of something like an `if` statement to perform work. We'll talk about this in more detail, but if there are only two possibilities this should probably just look like
```
if <clause>
  <do thing>
else
  <do other thing>
end
```